File: SOC/src/cc/game/soc/core/Island.java
package cc.game.soc.core;

import java.util.ArrayList;
import java.util.List;

import cc.lib.utils.Reflector;

public class Island extends Reflector<Island> {

	static {
		addAllFields(Island.class);
	}
	
	public Island() {
		this(0);
	}
	
	Island(int num) {
		this.num = num;
	}
	
	int num; // starts at 1
	final List<Integer> tiles = new ArrayList<Integer>();
	final List<Integer> borderRoute = new ArrayList<Integer>();
	final boolean [] discovered = new boolean[16];
	
	public int getNum() {
		return num;
	}
	
	public Iterable<Integer> getTiles() {
		return tiles;
	}
	
	public Iterable<Integer> getShoreline() {
		return borderRoute;
	}
	
	public boolean isDiscoveredBy(int playerNum) {
		if (playerNum > 0 && playerNum < discovered.length) {
			return discovered[playerNum];
		}
		return false;
	}
}

File: SOC/src/cc/game/soc/core/Island.kt
package cc.game.soc.core

import cc.lib.utils.Reflector
import java.util.*

class Island internal constructor(  // starts at 1
	var num: Int) : Reflector<Island>() {
	companion object {
		init {
			addAllFields(Island::class.java)
		}
	}

	constructor() : this(0) {}

	@JvmField
    val tiles: MutableList<Int> = ArrayList()
	@JvmField
    val borderRoute: MutableList<Int> = ArrayList()
	@JvmField
    val discovered = BooleanArray(16)
	fun getTiles(): Iterable<Int> {
		return tiles
	}

	val shoreline: Iterable<Int>
		get() = borderRoute

	fun isDiscoveredBy(playerNum: Int): Boolean {
		return if (playerNum > 0 && playerNum < discovered.size) {
			discovered[playerNum]
		} else false
	}
}